handlers/localeCode/handler.go

```diff
@@ -0,0 +1,41 @@
+package localeCode
+
+import (
+	"context"
+	"net/http"
+
+	netHttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/log.go/log"
+)
+
+// CheckHeaderValueAndForwardWithRequestContext is a wrapper which adds a localeCode from the request header to context if one does not yet exist
+func CheckHeaderValueAndForwardWithRequestContext(h http.Handler) http.Handler {
```

Comment:
It seems like this is a very similar function to the other */handler.go files - possible make generic handler.go that contains this function that can takes in a header value from an enumerated whitelist?

Then the method here can call the generic function with a header.

For example localCode becomes value and netHttp.LocaleHeaderKey just becomes Key

Reply:
And a similar situation happens with `CheckCookieValueAndForwardWithRequestContext`. I will do the same.

Reply:
And again, for collectionID, for both headers and cookies.